File: ruby/intp/test/test_ast.rb
require_relative 'helper'

class AstTest < Minitest::Test
  def test_string
    let_stmt = Intp::LetStatement.new
    let_stmt.token = Intp::Token.new(Intp::Token::LET, 'let')
    let_stmt.name = Intp::Identifier.new(
      Intp::Token.new(Intp::Token::IDENT, 'myVar'),
      'myVar'
    )
    let_stmt.value = Intp::Identifier.new(
      Intp::Token.new(Intp::Token::IDENT, 'anotherVar'),
      'anotherVar'
    )
    program = Intp::Program.new([let_stmt])

    assert_equal program.to_s, 'let myVar = anotherVar;'
  end
end

File: python/tests/test_ast.py
import unittest
from intp import ast
from intp import token


class TestString(unittest.TestCase):
    def test_string(self):
        program = ast.Program()

        statement = ast.LetStatement(token.Token(token.LET, "let"))
        statement.name = ast.Identifier(token.Token(token.IDENT, "myVar"), "myVar")
        statement.value = ast.Identifier(token.Token(token.IDENT, "anotherVar"), "anothervar")
        program.statements = [statement]

        self.assertEqual(program.string(), "let myVar = anothervar;")
